tket/src/serialize/pytket/opaque/subgraph.rs

```diff
@@ -0,0 +1,177 @@
+//! Opaque subgraph definition.
+
+use hugr::ops::OpTrait;
+use itertools::{Either, Itertools};
+
+use std::collections::BTreeSet;
+
+use crate::serialize::pytket::PytketEncodeError;
+use hugr::core::HugrNode;
+use hugr::hugr::views::sibling_subgraph::InvalidSubgraph;
+use hugr::hugr::views::SiblingSubgraph;
+use hugr::types::Signature;
+use hugr::{Direction, Hugr, HugrView, IncomingPort, OutgoingPort};
+
+/// A subgraph of nodes in the Hugr that could not be encoded as TKET1
+/// operations.
+///
+/// of subgraphs it can represent; in particular const edges and order edge are
+/// not allowed in the boundary.
+#[derive(Debug, Clone)]
+pub struct OpaqueSubgraph<N> {
```

Comment:
Should we consider upstreaming this? In general, `SiblingSubgraph` doesn't seem production ready 😅